src/main/java/net/fabricmc/loader/impl/entrypoint/EntrypointStorage.java

```diff
@@ -35,6 +35,7 @@
 public final class EntrypointStorage {
 	interface Entry {
 		<T> T getOrCreate(Class<T> type) throws Exception;
+		boolean isOptional();
```

Comment:
How is this "optional"? This is more like "evaluate immediately"

Reply:
The legacy initializers aren't type safe, so the code adds one Entry for each potential type (client/common/server - type isn't known at registration time). Instantiating the entry points resolves the actual type and filters for the required type, this filtering operation makes some of the entries optional.